src/components/network/network-table.js

```diff
@@ -15,27 +15,48 @@ genotet.NetworkTable = function(data) {
 };
 
 /**
+ * Triggers a jQuery event on the table.
+ * @param {string} eventType Type of event.
+ * @param {*=} opt_data Data to be sent via the event.
+ */
+genotet.NetworkTable.prototype.signal = function(eventType, opt_data) {
+  $(this).trigger('genotet.' + eventType, [opt_data]);
+};
+
+/**
  * Creates an incident edge table with the given data, inside a given
  * container.
  * @param {!jQuery} table Table element as container.
  * @param {!Array<{
+ *   id: string,
  *   source: string,
  *   target: string,
- *   weight: number,
+ *   weight: !Array<number>,
  *   added: boolean
  * }>} edges List of edges.
- *   added: Whether the edge has been added to the network
+ *   added: Whether the edge has been added to the network.
  */
 genotet.NetworkTable.prototype.create = function(table, edges) {
   var edgeIds = {};
-  this.data.edges.forEach(function(edge) {
+  this.data.network.edges.forEach(function(edge) {
     edgeIds[edge.id] = true;
   });
   edges.forEach(function(edge) {
     edge.added = edge.id in edgeIds;
   });
-  table.DataTable({
-    data: edges,
+  var edgesForTable = [];
+  edges.forEach(function(edge) {
+    edgesForTable.push({
+      id: edge.id,
+      source: edge.source,
+      target: edge.target,
+      added: edge.added,
+      weight: edge.weight[0],
+      originalWeight: edge.weight
```

Comment:
Because the weight to show in table and edge-info module is just weight[0], so I just pass weight[0] to edgesForTable.weight, and store the weight: !Array{number} in originalWeight. When adding an edge, I need to store the weight array in this.network.edges. And I can simply pick the weight array up from table, not need to query from server, those attributes like id and originalWeight are just stored in the table but not shown. Maybe I should change the variable names to make it clear.